src/main/scala/org/zalando/spearheads/innkeeper/routes/Routes.scala

```diff
@@ -33,36 +33,48 @@ class Routes @Inject() (
     metrics: RouteMetrics)(
     implicit
     val authService: AuthService,
-    implicit val executionContext: ExecutionContext) {
-
-  private val logger = LoggerFactory.getLogger(this.getClass)
+    implicit val executionContext: ExecutionContext) extends StrictLogging {
 
   val route: RequestContext => Future[RouteResult] =
-    handleRejections(rejectionHandler()) {
-      extractRequest { req =>
-        val reqDesc = s"${req.method.toString()} ${req.uri.path.toString()}"
-
-        authenticationToken(reqDesc) { token =>
-          authenticate(token, reqDesc) { authenticatedUser =>
-            logger.debug(s"$reqDesc AuthenticatedUser: $authenticatedUser")
+    extractMethod { requestMethod =>
```

Comment:
@eugenkiss told me that this might dsl specific and that's why the syntax, because i was just wondering if scala supports a nicer syntax for function composition that this